jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/PrecompressedHttpContent.java

```diff
@@ -139,12 +139,12 @@ public Map<CompressedContentFormat, HttpContent> getPrecompressedContents()
     @Override
     public ByteBuffer getBuffer()
     {
-        return _content.getBuffer();
+        return _precompressedContent.getBuffer();
     }
 
     @Override
     public void release()
     {
-        _content.release();
+        _precompressedContent.release();
```

Comment:
I don't know. The `_content` will be accessible separately to the pre-compressed version and will have separate entry in the cache, so maybe this would result in a double release?

At the moment it looks there are no implementations of `HttpContent.release()` which actually do anything, so this currently has no effect.